### PR TITLE
Fixes #19582 - Alert when puppetclass is not saved due to error

### PR DESCRIPTION
**Pull request by orrabin**
Getting an error from a smart class parameter or smart variable that was not edited and prevents saving is confusing.
This alert is to make sure it's clear the puppet class will not be saved if there are any errors in any of it's parameters/variables.

### CODE REVIEW

File: app/views/puppetclasses/_form.html.erb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <%= form_for @puppetclass, :url => (@puppetclass.new_record? ? puppetclasses_path : puppetclass_path(:id => @puppetclass)) do |f| %>
   <%= base_errors_for @puppetclass %>
+  <% unless @puppetclass.errors.empty? %>
```
Comment (orrabin) [inline app/views/puppetclasses/_form.html.erb:3]: should this be only for errors on parameters/variables?

**Comment by orrabin** [comment]
@Rohoover is this the correct place and wording of the inline notification?

image

**Comment by Rohoover** [comment]
@orrabin 
Can the error occur on any of the tabs? Or is it specifically in the Smart Variable Tab?

**Comment by orrabin** [comment]
> Can the error occur on any of the tabs? Or is it specifically in the Smart Variable Tab?

@Rohoover any of the tabs

**Comment by Rohoover** [comment]
@orrabin 

So I think the position is correct because the error affects the whole page.

File: app/views/puppetclasses/_form.html.erb [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+    <%= alert :class => 'alert-danger', :header => '', :text => _("The puppet class has not been saved because of an existing error in the class or one of it's parameters.") %>
```
Comment (ShimShtein) [inline app/views/puppetclasses/_form.html.erb]: nitpick: IMHO, it should be either `"Puppet class has not..."` or `"The class has not..."`.

**Review by ShimShtein — COMMENTED** [review]
@ohadlevy, you did some work with error notifications, should it affect this message in some way?

**Comment by orrabin** [comment]
@ShimShtein thanks, I fixed the wording.

File: app/views/puppetclasses/_form.html.erb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 <%= form_for @puppetclass, :url => (@puppetclass.new_record? ? puppetclasses_path : puppetclass_path(:id => @puppetclass)) do |f| %>
   <%= base_errors_for @puppetclass %>
+  <% unless @puppetclass.errors.empty? %>
+    <%= alert :class => 'alert-danger', :header => '', :text => _("The class has not been saved because of an existing error in the class or one of it's parameters.") %>
```
Comment (tbrisker) [inline app/views/puppetclasses/_form.html.erb]: nitpick: its not it's, also I would drop the word "existing"

**Review by dLobatog — CHANGES_REQUESTED** [review]
Set to WOC as per @tbrisker 's comment about the typo, I haven't tried it but it looks fine

**Comment by orrabin** [comment]
@tbrisker thanks, I fixed the typo.

File: app/views/puppetclasses/_form.html.erb (hunk 1 above)
Comment (tbrisker) [reply app/views/puppetclasses/_form.html.erb:3]: I don't think there can be any error in the class itself, so if there is any error it has to be in the parameters/variables.

File: app/views/puppetclasses/_form.html.erb [hunk 4]
```diff
@@ -1,5 +1,8 @@
 <%= form_for @puppetclass, :url => (@puppetclass.new_record? ? puppetclasses_path : puppetclass_path(:id => @puppetclass)) do |f| %>
   <%= base_errors_for @puppetclass %>
+  <% unless @puppetclass.errors.empty? %>
+    <%= alert :class => 'alert-danger', :header => '', :text => _("The class has not been saved because of an error in the class or one of its parameters.") %>
```
Comment (tbrisker) [inline app/views/puppetclasses/_form.html.erb]: could you reword this to `The class could not be saved because of an error in one of the class parameters.`?

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
Thanks @orrabin !
